src/librustc/hir/lowering.rs

```diff
@@ -1833,8 +1835,12 @@ impl<'a> LoweringContext<'a> {
                 ExprKind::Try(ref sub_expr) => {
                     // to:
                     //
+                    // #[allow(unreachable_patterns)]
```

Comment:
I still seem to get warnings without these. At least the `unreachable_code` attribute seems to have an effect. I was thinking I'll change the code to this:

```rust
match Carrier::translate(<expr>) {
    #[allow(unused)]
    Ok(val) => val,
    #[allow(unused)]
    Err(err) => return Carrier::from_error(From::from(err)),
}
```